insurance_policy.go

```diff
@@ -0,0 +1,233 @@
+package elation
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type InsurancePolicyServicer interface {
+	Create(ctx context.Context, patientID int64, create *InsurancePolicyCreate) (*InsurancePolicy, *http.Response, error)
+	Find(ctx context.Context, patientID int64, opts *FindInsurancePoliciesOptions) (*InsurancePolicyResponse, *http.Response, error)
+	Get(ctx context.Context, patientID int64, id int64) (*InsurancePolicy, *http.Response, error)
+	Update(ctx context.Context, patientID int64, id int64, update *InsurancePolicy) (*InsurancePolicy, *http.Response, error)
+	Delete(ctx context.Context, patientID int64, id int64) (*http.Response, error)
+}
+
+var _ InsurancePolicyServicer = (*InsurancePolicyService)(nil)
+
+type InsurancePolicyService struct {
+	client *HTTPClient
+}
+
+type InsurancePolicyResponse struct {
+	// TODO: Determine how the pagination fields here work / if they work yet
+
+	Results []*InsurancePolicy `json:"results"`
+}
+
+type InsurancePolicy struct {
+	ID int64 `json:"id"`
+
+	PracticeID       int64  `json:"practice_id"`
+	PatientID        int64  `json:"patient_id"`
+	PatientFirstName string `json:"patient_first_name"`
+	PatientLastName  string `json:"patient_last_name"`
+	PatientDOB       string `json:"patient_dob"` // Format: YYYY-MM-DD
+
+	Status      string  `json:"status"` // enum: "active", "inactive"
+	Rank        *int64  `json:"rank"`   // enum: 1=primary, 2=secondary, 3=tertiary
+	CarrierID   *int64  `json:"carrier_id"`
+	CarrierName *string `json:"carrier_name"`
+	PlanID      *int64  `json:"plan_id"`
+	PlanName    *string `json:"plan_name"`
+	GroupID     *string `json:"group_id"`
+	MemberID    *string `json:"member_id"`
+	Copay       *string `json:"copay"`
+	Deductible  *string `json:"deductible"`
```

Comment:
Optional, but for this and the below strings with examples in the docs it might be nice to document the formats from their examples?